source/concepts.tex

```diff
@@ -0,0 +1,1249 @@
+  template <class B>
+  concept Boolean = @\seebelow@;
+
+  // \ref{concepts.lib.compare.equalitycomparable}, EqualityComparable:
+  template <class T>
+  concept EqualityComparable = @\seebelow@;
+
+  template <class T, class U>
+  concept EqualityComparableWith = @\seebelow@;
+
+  // \ref{concepts.lib.compare.stricttotallyordered}, StrictTotallyOrdered:
+  template <class T>
+  concept StrictTotallyOrdered = @\seebelow@;
+
+  template <class T, class U>
+  concept StrictTotallyOrderedWith = @\seebelow@;
+
+  // \ref{concepts.lib.object}, object concepts:
+  // \ref{concepts.lib.object.movable}, Movable:
+  template <class T>
+  concept Movable = @\seebelow@;
+
+  // \ref{concepts.lib.object.copyable}, Copyable:
+  template <class T>
+  concept Copyable = @\seebelow@;
+
+  // \ref{concepts.lib.object.semiregular}, Semiregular:
+  template <class T>
+  concept Semiregular = @\seebelow@;
+
+  // \ref{concepts.lib.object.regular}, Regular:
+  template <class T>
+  concept Regular = @\seebelow@;
+
+  // \ref{concepts.lib.callable}, callable concepts:
+  // \ref{concepts.lib.callable.invocable}, Invocable:
+  template <class F, class... Args>
+  concept Invocable = @\seebelow@;
+
+  // \ref{concepts.lib.callable.regularinvocable}, RegularInvocable:
+  template <class F, class... Args>
+  concept RegularInvocable = @\seebelow@;
+
+  // \ref{concepts.lib.callable.predicate}, Predicate:
+  template <class F, class... Args>
+  concept Predicate = @\seebelow@;
+
+  // \ref{concepts.lib.callable.relation}, Relation:
+  template <class R, class T, class U>
+  concept Relation = @\seebelow@;
+
+  // \ref{concepts.lib.callable.strictweakorder}, StrictWeakOrder:
+  template <class R, class T, class U>
+  concept StrictWeakOrder = @\seebelow@;
+}
+\end{codeblock}
+
+\rSec1[concepts.lib.corelang]{Core language concepts}
+
+\rSec2[concepts.lib.corelang.general]{General}
+
+\pnum
+This section contains the definition of concepts corresponding to language features.
+These concepts express relationships between types, type classifications, and
+fundamental type properties.
+
+\pnum
+\begin{example}
+Several concept specifications in this subclause are specified in terms of a type
+trait, and include an annotation that there need be no subsumption relationship
+between the concept and the type trait. Given the following set of declarations:
+\begin{codeblock}
+template <class T> concept C1 = is_c1_v<T>;
+template <class U> concept C2 = C1<U> && true;
+
+template <C1 T> void f(T);     // \#1
+template <C2 T> void f(T);     // \#2
+
+template <class T>
+void g(T) requires is_c1_v<T>; // \#3
+template <C2 T> void g(T);     // \#4
+\end{codeblock}
+Since \tcode{C2} subsumes \tcode{C1}, the expression \tcode{f(42)} unambiguously
+calls \#2. \tcode{g(42)}, however, is potentially ambiguous if there need be no
+subsumption relationship between \tcode{C1} and \tcode{is_c1_v}.
+\end{example}
+
+\rSec2[concepts.lib.corelang.same]{Concept \tcode{Same}}
```

Comment:
In general, let's try to make the labels a little shorter. Just "concepts.same" or ... singular "concept.same" seems sufficient. (Yes, rplease ename all labels from concepts.... to concept...)

Reply:
I'm re-stable-naming all of the specific concept subclauses to "concept.name", and leaving the general and collection stable names as "concepts.syn", "concepts.general", "concepts.compare".